Make the map type inaccessible to users (#2976)
tenzir/vast: web/docs/setup/monitor.md

      real-time: false
      path: /tmp/vast-metrics.sock
      type: datagram # possible values are "stream" or "datagram"
    # Configures if and where metrics should be written to VAST itself.
    self-sink:
      enable: false
```

Both the file and UDS sinks write metrics as NDJSON and inlines the `metadata`
key-value pairs into the top-level object. VAST buffers metrics for these sinks

> in order to batch I/O operations. To enable real-time metrics reporting, set the

Reviewer: ```suggestion
to batch I/O operations. To enable real-time metrics reporting, set the
```

Author: Fixed this in https://github.com/tenzir/vast/pull/2976/commits/f594483ae0a48c67bb844cee77dc7294b781aaf6 already.

`vast.metrics.file-sink.real-time` or `vast.metrics.uds-sink.real-time`
respectively in your configuration file.

> :::tip Self Sink ❤️ Pipelines

Reviewer: The spacing around ❤️ is different here. </bikeshed off>

Author: Fixed this in https://github.com/tenzir/vast/pull/2976/commits/f594483ae0a48c67bb844cee77dc7294b781aaf6 already.

> The self-sink routes metrics as events into VAST's internal storage engine,

Reviewer: NB: "self sink" is a terrible name, but I don't have a better one at hand.

you to work with metrics using VAST's pipelines. The schema for the self-sink is
slightly different, with the key being embedded in the schema name:

```yaml
vast.metrics.<key>:
  record:
    - ts: timestamp
    - version: string
    - actor: string
    - key: string
    - value: string
    - <metadata...>
```

> Here's an example that queries for the start up latency of VAST's stores,

Reviewer: ```suggestion
Here's an example that shows the start up latency of VAST's stores,
```

> grouping into 10 second buckets and looking at the minimum and the maximum

Reviewer: ```suggestion
grouped into 10 second buckets and looking at the minimum and the maximum
```

latency, respectively, for all buckets.

```bash
vast export json '#type == "vast.metrics.passive-store.init.runtime"

>   | select ts,value

Reviewer: ```suggestion
  | select ts, value
```
